components/esp_lvgl_port/src/lvgl8/esp_lvgl_port_disp.c

```diff
@@ -415,19 +451,19 @@ static void lvgl_port_flush_callback(lv_disp_drv_t *drv, const lv_area_t *area,
     lv_color_t *to = NULL;
 
     if (disp_ctx->trans_size == 0) {
-        if (disp_ctx->disp_type == LVGL_PORT_DISP_TYPE_RGB && (drv->direct_mode || drv->full_refresh)) {
+        if ((disp_ctx->disp_type == LVGL_PORT_DISP_TYPE_RGB || disp_ctx->disp_type == LVGL_PORT_DISP_TYPE_DSI) && (drv->direct_mode || drv->full_refresh)) {
             if (lv_disp_flush_is_last(drv)) {
                 /* If the interface is I80 or SPI, this step cannot be used for drawing. */
                 esp_lcd_panel_draw_bitmap(disp_ctx->panel_handle, x_start, y_start, x_end + 1, y_end + 1, color_map);
                 /* Waiting for the last frame buffer to complete transmission */
-                ulTaskNotifyValueClear(NULL, ULONG_MAX);
-                ulTaskNotifyTake(pdTRUE, portMAX_DELAY);
+                xSemaphoreTake(disp_ctx->trans_sem, 0);
```

Comment:
Please, could you update this with wait callback like this: https://github.com/espressif/esp-bsp/blob/61ab9cb60d5206962a563cb1caaf4a291d40bb80/components/esp_lvgl_port/src/lvgl9/esp_lvgl_port_disp.c#L476 (latest master)

Reply:
Sure, I can add an interface. Additionally, I want to explain that the change to a semaphore is due to certain scenarios. For example, when using a camera with LVGL, to prevent screen tearing, the camera needs to call lv_refr_now(). With the previous approach, it would cause the LVGL task to hang.

Reply:
Ok, I understand. But with wait callback it can be faster. LVGL can continue work during waiting and task is not hanged.

Reply:
Yes, the original method would indeed be faster, but from my testing, I found that the impact isn't significant. In the future, the ESP32-P4 development board may release some AI features integrated with LVGL, such as facial recognition. For now, this feature is still needed for support.

Reply:
In fact, I believe the lvgl_port_flush_wait_callback interface should not be used. Before calling ulTaskNotifyValueClear(NULL, ULONG_MAX) and ulTaskNotifyTake(pdTRUE, portMAX_DELAY), esp_lcd_panel_draw_bitmap should be called first to ensure buffer switching can occur. (Calling esp_lcd_panel_draw_bitmap will not immediately switch the buffer, but will wait until the current frame is completely flushed before switching.) Otherwise, rendering and reading might occur on the same buffer.